### PR TITLE
Viewusers

### PR DESCRIPTION
**Pull request by ahmad1091**
##### also, I build view_users route which contains:
* tow get routes (get ,userIndex).
* tow post routes(post ,authorization).
all user controllers placed in users.js
all user queries placed in users.js in database>queries file

### CODE REVIEW

File: config.env [hunk 1]
```diff
@@ -1,2 +1,3 @@
-DB_URL = postgres://acmt:12345678@localhost:5432/acmt
-SECRET = AbuMohammed
+DB_URL = postgres://four:123@localhost:5432/four
+TEST_DB_URL = postgres://fourtest:123@localhost:5432/fourtest
```
Comment (MMortaga) [inline config.env]: Are you sure you added `config.env` to `.gitignore`?

File: src/controllers/users.js [hunk 2]
```diff
@@ -21,21 +71,45 @@ exports.post = (request, response, next) => {
           if (!res.rows[0]) {
             checkId(idNumber).then((result) => {
               if (!result.rows[0]) {
-                addUser(data, hash, role)
-                  .then(() => {
-                    response.send({ message: 'User has been added successsfully' });
-                  }).catch((err) => { next(err); });
+                checkMobile(mobile).then((resul) => {
```
Comment (MMortaga) [inline src/controllers/users.js]: can you change `resul` to `mobileResult`? We need to keep our code as clean as possible.

File: src/database/queries/users.js [hunk 3]
```diff
@@ -27,8 +34,37 @@ const checkId = (idNumber) => {
...
+    text: 'SELECT mobile FROM users WHERE mobile=$1',
+    values: [mobile],
+  };
+  return dbConnection.query(sql);
+};
+
+const updateRole = (data) => {
+  const { role, userName } = data;
+  const sql = {
+    text: 'UPDATE users SET role=$1 where username =$2',
```
Comment (MMortaga) [inline src/database/queries/users.js]: Capitalize `where` :)

File: src/database/queries/users.js [hunk 4]
```diff
@@ -27,8 +34,37 @@ const checkId = (idNumber) => {
...
+    values: [role, userName],
+  };
+  return dbConnection.query(sql);
+};
+
+const deleteUser = (userName) => {
+  const sql = {
+    text: 'delete from users where username =$1',
```
Comment (MMortaga) [inline src/database/queries/users.js]: Don't forget to capitalize your keywords here, too.

File: config.env (hunk 1 above)
Comment (ahmad1091) [reply config.env]: Yes, I'm very sure I don't know what's wrong.

File: src/controllers/users.js (hunk 2 above)
Comment (ahmad1091) [reply src/controllers/users.js]: Done

File: src/database/queries/users.js (hunk 3 above)
Comment (ahmad1091) [reply src/database/queries/users.js]: Done

File: src/database/queries/users.js (hunk 4 above)
Comment (ahmad1091) [reply src/database/queries/users.js]: Done

File: config.env (hunk 1 above)
Comment (MMortaga) [reply config.env]: I'll remove the environmental variables from the web editor, please don't let them come through again.

**Pull request merged**